modules/photon/photon.js

```diff
@@ -1,55 +1,61 @@
 /* jshint onevar: false */
 
-(function($){
+( function() {
+	function recalculate() {
+		if ( this.complete ) {
+			// Support for lazy loading: if there is a lazy-src attribute and it's value
+			// is not the same as the current src we should wait until the image load event
+			var lazySrc = this.getAttribute('data-lazy-src');
+			if ( lazySrc && this.src !== lazySrc ) {
+				this.addEventListener( 'load', recalculate );
+				return;
+			}
+
+			// Copying CSS width/height into element attributes.
+			var width = this.width;
+			var height = this.height;
+			if ( width && width > 0 && height && height > 0 ) {
+				this.setAttribute('width', width);
+				this.setAttribute('height', height);
+
+				reset_for_retina( this );
+			}
+		}
+		else {
+			this.addEventListener( 'load', recalculate );
+			return;
+		}
+	}
+
 	/**
 	 * For images lacking explicit dimensions and needing them, try to add them.
 	 */
 	var restore_dims = function() {
-		$( 'img[data-recalc-dims]' ).each( function recalc() {
-			var $this = $( this );
-			if ( this.complete ) {
-
-				// Support for lazy loading: if there is a lazy-src
-				// attribute and it's value is not the same as the current src we
-				// should wait until the image load event
-				if ( $this.data( 'lazy-src' ) && $this.attr( 'src' ) !== $this.data( 'lazy-src' ) ) {
-					$this.load( recalc );
-					return;
-				}
-
-				var width = this.width,
-					height = this.height;
-
-				if ( width && width > 0 && height && height > 0 ) {
-					$this.attr( {
-						width: width,
-						height: height
-					} );
-
-					reset_for_retina( this );
-				}
-			}
-			else {
-				$this.load( recalc );
-			}
-		} );
+		var elements = document.querySelectorAll( 'img[data-recalc-dims]' );
+		for (var i = 0; i < elements.length; i++) {
+			recalculate.call( elements[i] );
+		}
 	},
 
 	/**
 	 * Modify given image's markup so that devicepx-jetpack.js will act on the image and it won't be reprocessed by this script.
 	 */
 	reset_for_retina = function( img ) {
-		$( img ).removeAttr( 'data-recalc-dims' ).removeAttr( 'scale' );
+		img.removeAttribute( 'data-recalc-dims' );
+		img.removeAttribute( 'scale' );
 	};
 
+	// Vanilla version of jQuery.ready()
+	var ready = function( fn ) {
+		if ( document.attachEvent ? document.readyState === 'complete' : document.readyState !== 'loading' ) {
+			fn();
+		} else {
+			document.addEventListener( 'DOMContentLoaded', fn );
+		}
+	};
 	/**
 	 * Check both when page loads, and when IS is triggered.
 	 */
-	$( document ).ready( restore_dims );
-
-	if ( 'on' in $.fn ) {
```

Comment:
This condition only makes sense for jQuery which is basically a fallback for old jQuery versions